hbase_storage/src/main/java/com/splicemachine/access/client/SkeletonClientSideRegionScanner.java

```diff
@@ -189,8 +232,7 @@ public HRegion getRegion(){
     /*private helper methods*/
 
     private boolean updateTopCell(boolean response, List<Cell> results) throws IOException {
-        if (!results.isEmpty() &&
-                CellUtil.matchingFamily(results.get(0),ClientRegionConstants.FLUSH)){
+        if (matchingFamily(results, ClientRegionConstants.FLUSH)) {
             if (LOG.isDebugEnabled())
                 SpliceLogUtils.debug(LOG,"received flush message " + results.get(0));
             flushed = true;
```

Comment:
I think this is neither thread-safe nor exception-safe.
Instead maybe we should:
1. Moved the `flushed` flag into the `updateScanner` method and update it there with care.
2. Mark `updateScanner` method as `synchronized`.

Reply:
It does not need to be thread safe, only one thread is using this scanner.